eZ/Publish/SPI/Repository/Tests/Decorator/ContentServiceDecoratorTest.php

```diff
@@ -0,0 +1,484 @@
+        ];
+
+        $serviceMock->expects($this->exactly(1))->method('removeTranslation')->with(...$parameters);
+
+        $decoratedService->removeTranslation(...$parameters);
+    }
+
+    public function testDeleteTranslationDecorator()
+    {
+        $serviceMock = $this->createServiceMock();
+        $decoratedService = $this->createDecorator($serviceMock);
+
+        $parameters = [
+            $this->createMock(ContentInfo::class),
+            'random_value_5ced05ce156d02.84155908',
+        ];
+
+        $serviceMock->expects($this->exactly(1))->method('deleteTranslation')->with(...$parameters);
+
+        $decoratedService->deleteTranslation(...$parameters);
+    }
+
+    public function testDeleteTranslationFromDraftDecorator()
+    {
+        $serviceMock = $this->createServiceMock();
+        $decoratedService = $this->createDecorator($serviceMock);
+
+        $parameters = [
+            $this->createMock(VersionInfo::class),
+            'random_value_5ced05ce156d37.22902273',
+        ];
+
+        $serviceMock->expects($this->exactly(1))->method('deleteTranslationFromDraft')->with(...$parameters);
+
+        $decoratedService->deleteTranslationFromDraft(...$parameters);
+    }
+
+    public function testHideContentDecorator()
+    {
+        $serviceMock = $this->createServiceMock();
+        $decoratedService = $this->createDecorator($serviceMock);
+
+        $parameters = [$this->createMock(ContentInfo::class)];
+
+        $serviceMock->expects($this->exactly(1))->method('hideContent')->with(...$parameters);
+
+        $decoratedService->hideContent(...$parameters);
+    }
+
+    public function testRevealContentDecorator()
+    {
+        $serviceMock = $this->createServiceMock();
+        $decoratedService = $this->createDecorator($serviceMock);
+
+        $parameters = [$this->createMock(ContentInfo::class)];
+
+        $serviceMock->expects($this->exactly(1))->method('revealContent')->with(...$parameters);
+
+        $decoratedService->revealContent(...$parameters);
+    }
+
+    public function testNewContentCreateStructDecorator()
+    {
+        $serviceMock = $this->createServiceMock();
+        $decoratedService = $this->createDecorator($serviceMock);
+
+        $parameters = [
+            $this->createMock(ContentType::class),
+            'random_value_5ced05ce156db7.87562997',
+        ];
+
+        $serviceMock->expects($this->exactly(1))->method('newContentCreateStruct')->with(...$parameters);
+
+        $decoratedService->newContentCreateStruct(...$parameters);
+    }
+
+    public function testNewContentMetadataUpdateStructDecorator()
+    {
+        $serviceMock = $this->createServiceMock();
+        $decoratedService = $this->createDecorator($serviceMock);
+
+        $parameters = [];
+
+        $serviceMock->expects($this->exactly(1))->method('newContentMetadataUpdateStruct')->with(...$parameters);
+
+        $decoratedService->newContentMetadataUpdateStruct(...$parameters);
```

Comment:
Sadly this triggers inspection of invalid number of parameters used by the method call, so I think if method has no parameters, we should drop passing them.
This can be detected by running code inspections for all decorators in PhpStorm.

Reply:
Sounds fair. I'll double check no-parameters cases.